pyfr/integrators/dual/pseudo/base.py

```diff
@@ -35,6 +35,9 @@ def __init__(self, backend, systemcls, rallocs, mesh,
             raise TypeError('Incompatible pseudo-stepper/pseudo-controller '
                             'combination')
 
+        # Ensure the system is compatible with our formulation
+        systemcls.elementscls.validate_formulation(self.formulation, self, cfg)
```

Comment:
Looking at this in the flesh there is no need to pass formulation or cfg; just pass self since both the cfg and formulation are public attributes.